### ISSUE TITLE
Performance Issues when using WMS Filters

### ISSUE THREAD

**Issue by geographika** [MapServer/MapServer#6258]
WMS filters were added in [RFC 118](https://mapserver.org/development/rfc/ms-rfc-118.html). Allowing WMS requests to be filtered, for example:

```
mapserv -nh "QUERY_STRING=map=my.map&SERVICE=WMS&VERSION=1.3.0&REQUEST=GetMap&FORMAT=image/png&TRANSPARENT=true&STYLES=Type&TILED=false&LAYERS=MyLayer&ORDERBY=StartTime&FILTER=(<Filter><PropertyIsEqualTo><PropertyName>MyField</PropertyName><Literal>My Property</Literal></PropertyIsEqualTo></Filter>)&CRS=EPSG:3857&WIDTH=1571&HEIGHT=843&BBOX=-1230789.0294479083,6622962.752797382,-750458.7437038606,6880708.412174996"
```

For a typical WMS request the `msLayerWhichItems` function gathers all the fields required to display the WMS e.g. for classes, labels, filters etc. and does not request any additional fields.

https://github.com/MapServer/MapServer/blob/20b7671f788952c6097ae154b667dd564cf70764/maplayer.c#L873

When a `FILTER` is added then *all* fields for the layer are requested, even though they may not be used:

https://github.com/MapServer/MapServer/blob/2bc7f625d7ec45d8401aeb66ae908b5210acf83f/mapquery.c#L835

I have a layer with ~50 fields. When displaying with just the required fields it takes < 1s. When all fields are requested it takes over 6 minutes. This is a similar issue to #6131 for WFS. 

I propose that if in the `msWMSApplyFilter` a metadata property is set so that returning all fields can be skipped later. Trying to pass a flag from this function would have to be passed through 6 functions to reach to the `msQueryByFilter` so modifying the metadata would be less intrusive. `msWMSApplyFilter` is not called when using `GetFeatureInfo`, so these requests would be unaffected (and continue to return all fields, which is less of an issue as this typically only has to query a handful of features). 

https://github.com/MapServer/MapServer/blob/207594b6685047269d8c2f1289c26644eec36787/mapwms.cpp#L303

**Comment by geographika** [comment]
I've been running this on staging servers for a couple of weeks without issues (apart from a huge performance boost). 
If there are no objections I'd like to merge this soon.

**Comment by jratike80** [comment]
+1 for the boost.

**Comment by geographika** [comment]
Thanks all for the feedback/comments! Closed with #6259
